software_training/src/action.cpp

```diff
@@ -0,0 +1,49 @@
+#include <ros/ros.h>
+#include <actionlib/client/simple_action_client.h>
+#include <actionlib/>
+#include "std_msgs/String.h"
+#include "geometry_msgs/Twist.h"
+#include <sstream>
+#include <cmath>
+#include "software_training/Distance.h"
+#include <actionlib/server/simple_action_server.h>
+
+int main (int argc, char **argv)
+{
+  ros::init(argc, argv, "action");
+  
+  ros::NodeHandle nh;
+  ros::Publisher pub
+       = nh.advertise<geometry_msgs::Twist>("/moving_turtle/cmd_vel", 1000);
+  // create the action client
+  // true causes the client to spin its own thread
+
+  actionlib::SimpleActionClient<goemetry_msgs::Twist>("action", true);
+
+  Server server(n, "action", boost::bind(&execute, _1, &server), false);
+  server.start();
+
+  ROS_INFO("Waiting for action server to start.");
+  // wait for the action server to start
+  ac.waitForServer(); //will wait for infinite time
```

Comment:
ac should be the simple action client, here its not defined so this won't compile